Setting-Up-Ubuntu-For-Programming.md

```diff
@@ -0,0 +1,219 @@
+# Setting up Ubuntu for Programming
+
+> Note: We will be referring to Ubuntu 16.04 LTS (Xenial Xerus) in this article.
+
+1. Download and Install Ubuntu Desktop
+2. Customising Ubuntu
+3. Jazzing up the Terminal
+4. Installing DevTools and modern web browsers
+5. Downloading Text Editors and IDEs
+6. Alternatives to popular Windows and Mac software
+
+## Download and Install Ubuntu Desktop
+
+You can download the latest version of Ubuntu Desktop [here](http://www.ubuntu.com/download/desktop).
+If you need any help with installing the operating system [this](http://www.ubuntu.com/download/desktop/install-ubuntu-desktop) tutorial by Ubuntu will help you out. If you want to dual boot your computer with Windows and Ubuntu checkout [this](https://help.ubuntu.com/community/WindowsDualBoot) helpful article.
+
+Once Ubuntu is installed, open the terminal and type `sudo apt-get update && sudo apt-get upgrade` to update your packages.
+
+## Customising Ubuntu
+
+#### `.deb` Package Installer
+
+Install gdebi (`.deb` package installer) by typing `sudo apt-get install gdebi`.
```

Comment:
In general, I prefer doing a `dpkg` install, because underneath the hoods, apt-get also uses `dpkg`. If a dependency is missing for some `.deb` package, `sudo dpkg install -f` would fix that and complete the installation.

But as discussed in our call yesterday, let's not nitpick this so closely. We can always file new PRs to improve upon existing articles.